### PR TITLE
🎉  Support git.corp

### PR DESCRIPTION
**Pull request by macdonst**
## Description

In order to support git.corp list of contributors I needed to add the path to the users avatar to the graphql node.

## Related Issue

n/a

## Motivation and Context

Add support for internal git repos.

## How Has This Been Tested?

Working in client template, tested with gatsby develop and gatsby build.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by macdonst** [comment]
@shazron still working on this but wanted to get your eyes on it overnight my time. Does this make sense to you so we can support both git.corp and github.

File: gatsby-node.js [hunk 1]
```diff
@@ -60,10 +79,20 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
   for (const _path of paths) {
     let githubPath = path.join(root, _path.replace(process.cwd(), ''))
     if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
+    if (prefix && githubPath.startsWith(prefix)) {
+      githubPath = githubPath.slice(prefix.length + 1)
```
Comment (shazron) [inline gatsby-node.js]: This would remove one extra character (so the prefix entirely plus the next character after). Is this the intent

**Review by shazron — CHANGES_REQUESTED** [review]

File: gatsby-node.js (hunk 1 above)
Comment (macdonst) [reply gatsby-node.js]: Yeah it gets rid of the /. If I don't get rid of it the graph ql query fails.

**Review by shazron — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -37,21 +37,25 @@ plugins: [
...
           token: process.env.REPO_GITHUB_TOKEN, // Github Personal Access Token
           owner: process.env.REPO_OWNER, // user or org name
           name: process.env.REPO_NAME, 
           branch: process.env.REPO_BRANCH, // defaults to 'main'
-          default_branch: process.env.REPO_DEFAULT_BRANCH // defaults to 'main'
+          default_branch: process.env.REPO_DEFAULT_BRANCH, // defaults to 'main'
+          api: 'https://api.github.com/graphql' // API url, Optional. Default: 'https://api.github.com/graphql'
         }
       }
+    }
```
Comment (solimant) [inline README.md]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -37,21 +37,25 @@ plugins: [
...
   }
-];
+]
```
Comment (solimant) [inline README.md]:
```suggestion
];
```

File: README.md [hunk 4]
```diff
@@ -37,21 +37,25 @@ plugins: [
...
 ```
 
 GraphQL
+
```
Comment (solimant) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -61,17 +65,18 @@ GraphQL
       default_branch
       root
     }
-  }    
+  }
```
Comment (solimant) [inline README.md]:
```suggestion
  }    
```

File: README.md [hunk 6]
```diff
@@ -61,17 +65,18 @@ GraphQL
...
       contributors {
         date
         login
         name
+        avatarUrl
       }
       path
     }
-  }        
+  }
```
Comment (solimant) [inline README.md]:
```suggestion
  }        
```

File: gatsby-node.js [hunk 7]
```diff
@@ -31,12 +32,25 @@ exports.createSchemaCustomization = ({ actions }) => {
...
+    branch = 'main',
+    default_branch = 'main',
+    api = 'https://api.github.com/graphql'
+  } = options.repo ? options.repo : {}
 
   if (!token) {
-    console.warn('To get Github Contributors, a Github token is required (GITHUB_TOKEN environment variable)')
+    console.warn(
+      'To get Github Contributors, a Github token is required (GITHUB_TOKEN environment variable)'
+    )
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
    console.warn('To get Github Contributors, a Github token is required (GITHUB_TOKEN environment variable)')
```

File: gatsby-node.js [hunk 8]
```diff
@@ -45,7 +59,7 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
     expandDirectories: {
       extensions
     }
-  })
+  )
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
  })
```

File: src/gql.js [hunk 9]
```diff
@@ -70,7 +89,31 @@ async function githubFetchContributorsForPage (repoOwner, repoName, branch, page
...
+  ) {
+    console.log(
+      `The Github API didn't return the expected data, returning an empty contributor array. res: ${JSON.stringify(
+        res,
+        null,
+        2
+      )}`
+    )
+    return []
+  }
```
Comment (solimant) [inline src/gql.js]:
```suggestion
  `, token)
```

File: src/gql.js [hunk 10]
```diff
@@ -95,20 +138,23 @@ async function githubFetchContributorsForPage (repoOwner, repoName, branch, page
...
 
   // create a Set (thus unique items), by mapping via login
-  return Array.from(new Set(flattenedNodes.map(node => node.login)))
-    // map it back to the node (the first found will be the latest entry)
-    .map(login => flattenedNodes.find(node => node.login === login))
+  return (
+    Array.from(new Set(flattenedNodes.map((node) => node.login)))
+      // map it back to the node (the first found will be the latest entry)
+      .map((login) => flattenedNodes.find((node) => node.login === login))
+  )
```
Comment (solimant) [inline src/gql.js]:
```suggestion
  return Array.from(new Set(flattenedNodes.map(node => node.login)))
    // map it back to the node (the first found will be the latest entry)
    .map(login => flattenedNodes.find(node => node.login === login))
```

File: README.md [hunk 11]
```diff
@@ -13,7 +13,7 @@ governing permissions and limitations under the License.
 [![Version](https://img.shields.io/npm/v/@adobe/gatsby-source-github-file-contributors.svg)](https://npmjs.org/package/@adobe/gatsby-source-github-file-contributors)
 [![Downloads/week](https://img.shields.io/npm/dw/@adobe/gatsby-source-github-file-contributors.svg)](https://npmjs.org/package/@adobe/gatsby-source-github-file-contributors)
 [![Build Status](https://travis-ci.com/adobe/gatsby-source-github-file-contributors.svg?branch=master)](https://travis-ci.com/adobe/gatsby-source-github-file-contributors)
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) 
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
```
Comment (solimant) [inline README.md]:
```suggestion
[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) 
```

**Review by solimant — COMMENTED** [review]
Remove non-material changes

File: gatsby-node.js [hunk 12]
```diff
@@ -64,10 +76,24 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
   for (const _path of paths) {
     let githubPath = path.join(root, _path.replace(cwd, ''))
     if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
+    if (prefix && githubPath.startsWith(prefix)) {
+      githubPath = githubPath.slice(prefix.length + 1)
+    }
```
Comment (solimant) [inline gatsby-node.js:81]:
```suggestion
    let githubPath = _path.replace(root, '')

    if (githubPath.startsWith(path.sep)) {
      githubPath = githubPath.slice(1)
    }
```

**Review by solimant — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -37,14 +37,16 @@ plugins: [
         pages: {
           root: '', // root of the page paths (below) in the Github repo
           paths: ['src/pages'], // relative path of the pages from the config
-          extensions: ['md'] // page extensions to filter for
+          extensions: ['md'], // page extensions to filter for
+          prefix: '' // prefix to remove from path, Optional
```
Comment (solimant) [inline README.md:41]:
```suggestion
```

File: gatsby-node.js [hunk 14]
```diff
@@ -31,9 +32,20 @@ exports.createSchemaCustomization = ({ actions }) => {
...
 exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, options = {}) => {
   const cwd = process.cwd()
-  const root = options.root ? options.root : ''
-  const { paths: pages = ['src/pages'], extensions = ['md', 'mdx'] } = options.pages ? options.pages : {}
-  const { token, owner, name, branch = 'main', default_branch = 'main' } = options.repo ? options.repo : {}
+  const root = options.pages.root ? options.pages.root : options.root ? options.root : ''
+  const {
+    paths: pages = ['src/pages'],
+    extensions = ['md', 'mdx'],
+    prefix = ''
```
Comment (solimant) [inline gatsby-node.js:39]:
```suggestion
    extensions = ['md', 'mdx']
```

**Review by solimant — COMMENTED** [review]
Remove prefix

File: src/gql.js [hunk 15]
```diff
@@ -25,39 +24,59 @@ const gqlFetch = require('graphql-fetch')(GITHUB_GQL_ENDPOINT)
 /**
  * GraphQL fetch function.
  *
+ * @param {string} api the url of the GraphQL api
  * @param {string} query the GraphQL query
  * @param {string} token the Github Personal Access Token (repo scope only is needed)
  */
-async function githubFetch (query, token) {
+async function githubFetch(api, query, token) {
```
Comment (solimant) [inline src/gql.js]:
```suggestion
async function githubFetch (api, query, token) {
```

File: src/gql.js [hunk 16]
```diff
@@ -25,39 +24,59 @@ const gqlFetch = require('graphql-fetch')(GITHUB_GQL_ENDPOINT)
...
  * @param {string} repoOwner the Github org/owner to query from
  * @param {string} repoName the Github repo to query from
  * @param {string} branch the Github branch to query from
  * @param {string} pagePath the folder path for the pages in the repo to query from
  * @param {string} token the Github Personal Access Token
  * @returns {Array<GithubContributorInfo>} an array of the Github Contributor data
  */
-async function githubFetchContributorsForPage (repoOwner, repoName, branch, pagePath, token) {
-  const res = await githubFetch(`
+async function githubFetchContributorsForPage(
```
Comment (solimant) [inline src/gql.js]:
```suggestion
async function githubFetchContributorsForPage (
```

File: gatsby-node.js [hunk 17]
```diff
@@ -62,12 +73,26 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
   })
 
   for (const _path of paths) {
-    let githubPath = path.join(root, _path.replace(cwd, ''))
-    if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
+    let githubPath = _path.replace(root, '')
+
+    if (githubPath.startsWith(path.sep)) {
+      githubPath = githubPath.slice(1)
+    }
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
    let githubPath = path.join(root, _path.replace(cwd, ''))
    if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
    if (prefix && githubPath.startsWith(prefix)) {
      githubPath = githubPath.slice(prefix.length + 1)
    }
```

File: README.md [hunk 18]
```diff
@@ -37,14 +37,15 @@ plugins: [
         pages: {
           root: '', // root of the page paths (below) in the Github repo
           paths: ['src/pages'], // relative path of the pages from the config
-          extensions: ['md'] // page extensions to filter for
+          extensions: ['md'], // page extensions to filter for
```
Comment (solimant) [inline README.md:40]:
```suggestion
          extensions: ['md'], // page extensions to filter for
          prefix: '' // prefix to remove from path, Optional
```

File: gatsby-node.js [hunk 19]
```diff
@@ -31,9 +32,19 @@ exports.createSchemaCustomization = ({ actions }) => {
 
 exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, options = {}) => {
   const cwd = process.cwd()
-  const root = options.root ? options.root : ''
-  const { paths: pages = ['src/pages'], extensions = ['md', 'mdx'] } = options.pages ? options.pages : {}
-  const { token, owner, name, branch = 'main', default_branch = 'main' } = options.repo ? options.repo : {}
+  const root = options.pages.root ? options.pages.root : options.root ? options.root : ''
+  const {
+    paths: pages = ['src/pages'],
+    extensions = ['md', 'mdx']
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
    extensions = ['md', 'mdx'],
    prefix = ''
```

File: gatsby-node.js [hunk 20]
```diff
@@ -64,10 +76,24 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
   for (const _path of paths) {
     let githubPath = path.join(root, _path.replace(cwd, ''))
     if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
    if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
    console.log('githubPath.substr(1):', githubPath)
```

File: gatsby-node.js [hunk 21]
```diff
@@ -64,10 +76,24 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
...
+    if (prefix && githubPath.startsWith(prefix)) {
+      githubPath = githubPath.slice(prefix.length + 1)
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
      githubPath = githubPath.slice(prefix.length + 1)
      console.log('githubPath.slice(prefix.length + 1):', githubPath)
```

File: gatsby-node.js [hunk 22]
```diff
@@ -31,9 +32,20 @@ exports.createSchemaCustomization = ({ actions }) => {
 
 exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, options = {}) => {
   const cwd = process.cwd()
-  const root = options.root ? options.root : ''
-  const { paths: pages = ['src/pages'], extensions = ['md', 'mdx'] } = options.pages ? options.pages : {}
-  const { token, owner, name, branch = 'main', default_branch = 'main' } = options.repo ? options.repo : {}
+  const root = options.pages.root ? options.pages.root : options.root ? options.root : ''
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
  const root = options.pages
    ? options.pages.root
      ? options.pages.root
      : options.root
        ? options.root
        : ''
    : options.root
      ? options.root
      : ''
```

File: gatsby-node.js [hunk 23]
```diff
@@ -62,12 +74,32 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
...
+    console.log('root:', root)
+    console.log('_path:', _path)
     let githubPath = path.join(root, _path.replace(cwd, ''))
+    console.log('githubPath:', githubPath)
     if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
+    console.log('githubPath.substr(1):', githubPath)
+    if (prefix && githubPath.startsWith(prefix)) {
+      githubPath = githubPath.slice(prefix.length + 1)
+      console.log('githubPath.slice(prefix.length + 1):', githubPath)
+    }
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
    let githubPath = path.join(root, _path.replace(cwd, ''))

    if (prefix && githubPath.startsWith(prefix)) {
      githubPath = githubPath.replaceAll(prefix, '')
    }

    if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
```

File: gatsby-node.js [hunk 24]
```diff
@@ -62,12 +74,32 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
...
+          api,
+          owner,
+          name,
+          branch,
+          githubPath,
+          token
+        )
+      } catch (error) {
+        console.error(error)
+      }
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
      contributors = await githubFetchContributorsForPage(
        api,
        owner,
        name,
        branch,
        githubPath,
        token
      )
```

File: gatsby-node.js [hunk 25]
```diff
@@ -63,11 +83,23 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
 
   for (const _path of paths) {
     let githubPath = path.join(root, _path.replace(cwd, ''))
+
+    if (prefix && githubPath.startsWith(prefix)) {
+      githubPath = githubPath.replaceAll(prefix, '')
+    }
+
     if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
```
Comment (solimant) [inline gatsby-node.js:91]:
```suggestion
    if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)

    console.log('cwd:', cwd)
    console.log('prefix:', prefix)
    console.log('root:', root)
    console.log('githubPath:', githubPath)
```

File: gatsby-node.js [hunk 26]
```diff
@@ -63,11 +83,28 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
...
+    console.log('cwd:', cwd)
+    console.log('prefix:', prefix)
+    console.log('root:', root)
+    console.log('githubPath:', githubPath)
+
     let contributors = []
     if (token) {
-      contributors = await githubFetchContributorsForPage(owner, name, branch, githubPath, token)
+      contributors = await githubFetchContributorsForPage(
+        api,
```
Comment (solimant) [inline gatsby-node.js:101]:
```suggestion
        'https://git.corp.adobe.com/api/graphql',
```

File: gatsby-node.js [hunk 27]
```diff
@@ -63,11 +83,28 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
...
+    console.log('cwd:', cwd)
+    console.log('prefix:', prefix)
+    console.log('root:', root)
+    console.log('githubPath:', githubPath)
+
     let contributors = []
     if (token) {
-      contributors = await githubFetchContributorsForPage(owner, name, branch, githubPath, token)
+      contributors = await githubFetchContributorsForPage(
+        'https://git.corp.adobe.com/api/graphql',
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
        api,
```

File: README.md [hunk 28]
```diff
@@ -37,14 +37,16 @@ plugins: [
...
+          prefix: '' // prefix to remove from path, Optional
         },
         repo: {
           token: process.env.REPO_GITHUB_TOKEN, // Github Personal Access Token
           owner: process.env.REPO_OWNER, // user or org name
           name: process.env.REPO_NAME, 
           branch: process.env.REPO_BRANCH, // defaults to 'main'
-          default_branch: process.env.REPO_DEFAULT_BRANCH // defaults to 'main'
+          default_branch: process.env.REPO_DEFAULT_BRANCH, // defaults to 'main'
+          api: 'https://api.github.com/graphql' // API url, Optional. Default: 'https://api.github.com/graphql'
```
Comment (solimant) [inline README.md:49]:
```suggestion
          api: process.env.GH_GQL_API_ENDPOINT || 'https://api.github.com/graphql' // API url, Optional. Default: 'https://api.github.com/graphql'
```

File: README.md [hunk 29]
```diff
@@ -37,14 +37,16 @@ plugins: [
...
+          prefix: '' // prefix to remove from path, Optional
         },
         repo: {
           token: process.env.REPO_GITHUB_TOKEN, // Github Personal Access Token
           owner: process.env.REPO_OWNER, // user or org name
           name: process.env.REPO_NAME, 
           branch: process.env.REPO_BRANCH, // defaults to 'main'
-          default_branch: process.env.REPO_DEFAULT_BRANCH // defaults to 'main'
+          default_branch: process.env.REPO_DEFAULT_BRANCH, // defaults to 'main'
+          api: process.env.GH_GQL_API_ENDPOINT || 'https://api.github.com/graphql' // API url, Optional. Default: 'https://api.github.com/graphql'
```
Comment (solimant) [inline README.md]:
```suggestion
          api: 'https://api.github.com/graphql' // API url, Optional. Default: 'https://api.github.com/graphql'
```

File: gatsby-node.js [hunk 30]
```diff
@@ -63,11 +83,28 @@ exports.sourceNodes = async ({ actions, createNodeId, createContentDigest }, opt
...
+      githubPath = githubPath.replaceAll(prefix, '')
+    }
+
     if (githubPath.charAt(0) === path.sep) githubPath = githubPath.substr(1)
 
+    console.log('cwd:', cwd)
+    console.log('prefix:', prefix)
+    console.log('root:', root)
+    console.log('githubPath:', githubPath)
+
```
Comment (solimant) [inline gatsby-node.js]:
```suggestion
```

File: src/gql.js [hunk 31]
```diff
@@ -25,39 +24,59 @@ const gqlFetch = require('graphql-fetch')(GITHUB_GQL_ENDPOINT)
...
-    method: 'POST'
-  })
+  return gqlFetch(
+    query,
+    {},
+    {
+      headers,
+      method: 'POST'
+    }
+  )
```
Comment (solimant) [inline src/gql.js]:
```suggestion
  return gqlFetch(query, {}, {
    headers,
    method: 'POST'
  })
```

**Pull request merged**